libpod/runtime_img.go

```diff
@@ -155,9 +283,22 @@ func (r *Runtime) PullImage(imgName string, allTags bool, signaturePolicyPath st
 		if manifest.Annotations == nil || manifest.Annotations["org.opencontainers.image.ref.name"] == "" {
 			return errors.Errorf("error, archive doesn't have a name annotation. Cannot store image with no name")
 		}
-		images = append(images, manifest.Annotations["org.opencontainers.image.ref.name"])
-	} else {
-		images = append(images, imgName)
+		pullInfo, err := getPullStruct(srcRef, manifest.Annotations["org.opencontainers.image.ref.name"], r.store)
+		if err != nil {
+			return err
+		}
+		pullStructs = append(pullStructs, pullInfo)
+	} else if srcRef != nil && srcRef.Transport().Name() == DirTransport {
+		// supports pull from a directory
+		image := splitArr[1]
+		if image[:1] == "/" {
+			image = image[1:]
+		}
+		pullInfo, err := getPullStruct(srcRef, image, r.store)
+		if err != nil {
+			return err
+		}
+		pullStructs = append(pullStructs, pullInfo)
 	}
```

Comment:
Doesn’t this need a final ~`else if srcRef != nil {pullStructs= append(pullStructs, getPullStruct()}`? Otherwise `pullStructs` will be empty and nothing will happen.

(Also, the `srcRef`→`pullStructs` conversion code is big enough, and stand-alone enough, that it could be more readable as a separate function, with this something like
```go
	srcRef, err := alltransports.ParseImageName(imgName)
	if err != nil {
		// could be trying to pull from registry with short name
		pullStructs, err = getRegistriesToTry(imgName, r.store)
		if err != nil {
			fmt.Println("error getting default registries to try:", err) // or return errors.Wrap(), I guess
		}
	} else {
		// could be trying to pull from registry with short name
		pullStructs, err = getPullListFromRef(srcRef, r.store)
		if err != nil {
			// handle
		}
        }
```